### PR TITLE
Config flag to toggle between `vpcTransactionService` and `transactionService`

### LINKED ISSUE #629: Introduce `transactionService` and `vpcTransactionService` toggle env var
### Feature

From `v3.6.0` local development won't be possible as we would be using VPC internal URLs even in staging.

We need an easy way to use the external URL for the services.

### Proposal

Introduce env var `USE_EXTERNAL_TX_SERVICE_URI=<bool>` as a mean to toggle between using `transactionService` and `vpcTransactionService` URLs in the `core_uri!` macro.

### PR DESCRIPTION
**Pull request by jpalvarezl**
Closes #629 
Closes #631 

In addition to what @fmrsabino suggested, I introduced tests with debug assertion to also validate what is ran in the local setup unoptimized case.

However, after the discussion in the comments we have decided to stick to an implementation based on a environment variable, as this would allow better testability. 

Important note: From this PR onward tests would run in single thread, as changes to env vars behave non-deterministically in an async setup.

### CODE REVIEW

**Review by Uxio0 — COMMENTED** [review]
I would prefer to make it configurable via environment variable instead of compiling flags. Reasoning:
- More flexible and easy to use
- More visible. This feature could be shadowed in the future and at some point we might not know why the service behaves in a different way when testing locally

(cc @rmeissner  @fmrsabino)

**Comment by jpalvarezl** [comment]
I find myself inclined towards the `env var` impl too. As it would allow setting the env var per test, then we don't need to sacrifice testability. We can assert all execution paths without the need to have optimised and unoptimised builds simultaneously. 

Granted, this PR does not address this, we still only check `test` builds.

**Comment by jpalvarezl** [comment]
I added the env var implementation. However, I had to additionally make the tests run in a single thread, as the setting of the env var per test  was behaving non-deterministically, as our tests are `async` functions.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --verbose --all --locked
+          args: --verbose --all --locked -- --test-threads 1
```
Comment (rmeissner) [inline .github/workflows/rust.yml:61]: Why are these changes required?

**Review by rmeissner — COMMENTED** [review]
How does this relate to https://github.com/gnosis/safe-client-gateway/pull/631?  Would this replace that one?

File: .github/workflows/rust.yml (hunk 1 above)
Comment (rmeissner) [reply .github/workflows/rust.yml:61]: Ahhh ... didn't see the PR comment when reviewing ... could we add this to the description of the PR?

**Review by rmeissner — APPROVED** [review]

**Comment by jpalvarezl** [comment]
I understood #631 PR as a exploratory PR where @fmrsabino evaluated how that alternative would look like. If we merge this PR then we would have decided to go with the env var alternative.
